core/mutator/entry/mutation.go

```diff
@@ -156,16 +167,13 @@ func (m *Mutation) sign(signers []signatures.Signer) (*pb.Entry, error) {
 
 // Check verifies that an update was successfully applied.
 // Returns nil if newLeaf is equal to the entry in this mutation.
-func (m *Mutation) Check(newLeaf []byte) error {
+func (m *Mutation) Check(newLeaf []byte) (bool, error) {
```

Comment:
oops, I didn't see you've already thought about this since the previous commit.
Do consider the error constant alternative though.